client/state-hooks.ts

```diff
@@ -19,3 +19,29 @@ export function usePrevious<T>(value: T): T | undefined {
   })
   return ref.current
 }
+
+/**
+ * A hook to move a prop to a ref, so it can be used in `useCallback` and `useEffect` hooks, without
+ * needing to specify it in the dependency arrays of those hooks, thus reducing the amount of times
+ * those callbacks get recreated.
+ *
+ * @example
+ *
+ * export const CounterComponent = (props) => {
+ *   const countRef = usePropAsRef(props.count)
+ *   useEffect(() => {
+ *     if (countRef.current === 5) {
+ *       console.log('Count is 5!!!')
+ *     } else {
+ *       console.log('Count is not 5 :(')
+ *     }
+ *   }, [props.something])
```

Comment:
I'm not sure what this value is doing in this array but this would also be a lint warning since the closure would close over the entire `props` value. If you wrote this as actual code, the linter would ask to you to assign `something` to a variable rather than dereference it off of `props`. (But, since it is not used and is meaningless to the example, my preference is that you just delete it and pass an empty array)

Side note: I also don't think it's a particularly good example of the usefulness of this, a better example would be e.g. accessing a value in a click handler